src/components/approval-tasks/approval-notification/pipeline-approval-context.tsx

```diff
@@ -0,0 +1,176 @@
+import * as React from 'react';
+import { AlertVariant } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useSelector } from 'react-redux';
+import {
+  UserInfo,
+  WatchK8sResource,
+  useActiveNamespace,
+} from '@openshift-console/dynamic-plugin-sdk';
+import {
+  getGroupVersionKindForModel,
+  useK8sWatchResource,
+} from '@openshift-console/dynamic-plugin-sdk';
+import ApprovalToastContent from './ApprovalToastContent';
+import { ApprovalStatus, ApprovalTaskKind } from '../../../types';
+import { ApprovalTaskModel } from '../../../models';
+import { SDKStoreState } from '@openshift-console/dynamic-plugin-sdk/lib/app/redux-types';
+import { ApprovalLabels, ApprovalFields } from '../../../consts';
+import { useToast } from '../../toast/useToast';
+
+const getPipelineRunNameofEachApprover = (
+  approvalTask: ApprovalTaskKind,
+): string => {
+  let pipelineRunName = '';
+  if (
+    approvalTask?.metadata?.labels?.[
+      ApprovalLabels[ApprovalFields.PIPELINE_RUN]
+    ]
+  ) {
+    pipelineRunName =
+      approvalTask?.metadata?.labels?.[
+        ApprovalLabels[ApprovalFields.PIPELINE_RUN]
+      ];
+  } else {
+    pipelineRunName = approvalTask?.metadata?.name
+      ?.split('-')
+      ?.slice(0, -1)
+      ?.join('-');
+  }
+  return pipelineRunName;
+};
```

Comment:
Note to myself: must take a deeper look into this.

Reply:
Updated it. This function is no longer required. Else condition was not needed.